integration/ct_integration_test.go

```diff
@@ -517,35 +531,51 @@ func (want *wantStats) done(ep string, rc int) {
 }
 
 func (want *wantStats) check(cfg ctfe.LogConfig) error {
-	statsURI := "http://" + (*httpServerFlag) + "/debug/vars"
-	httpReq, err := http.NewRequest(http.MethodGet, statsURI, nil)
-	if err != nil {
-		return fmt.Errorf("failed to build GET request: %v", err)
-	}
-	client := new(http.Client)
+	ctx := context.Background()
+	got := newWantStats(int64(want.LogID))
+	rcs := []string{"200", "400"}
+	for _, s := range strings.Split(*httpServerFlag, ",") {
+		httpReq, err := http.NewRequest(http.MethodGet, "http://"+s+"/debug/vars", nil)
+		if err != nil {
+			return fmt.Errorf("failed to build GET request: %v", err)
+		}
+		client := new(http.Client)
 
-	httpRsp, err := ctxhttp.Do(context.Background(), client, httpReq)
-	if err != nil {
-		return fmt.Errorf("getting stats failed: %v", err)
-	}
-	defer httpRsp.Body.Close()
-	defer ioutil.ReadAll(httpRsp.Body)
-	if httpRsp.StatusCode != http.StatusOK {
-		return fmt.Errorf("got HTTP Status %q", httpRsp.Status)
-	}
+		httpRsp, err := ctxhttp.Do(ctx, client, httpReq)
+		if err != nil {
+			return fmt.Errorf("getting stats failed: %v", err)
+		}
+		defer httpRsp.Body.Close()
+		defer ioutil.ReadAll(httpRsp.Body)
+		if httpRsp.StatusCode != http.StatusOK {
+			return fmt.Errorf("got HTTP Status %q", httpRsp.Status)
+		}
+		var stats ctfe.AllStats
+		if err := json.NewDecoder(httpRsp.Body).Decode(&stats); err != nil {
+			return fmt.Errorf("failed to json.Decode() result: %v", err)
+		}
+		gotSingle := stats.Logs[cfg.Prefix]
+		if gotSingle.LogID != want.LogID {
+			return fmt.Errorf("got stats.log-id %d, want %d", got.LogID, want.LogID)
+		}
 
-	var stats ctfe.AllStats
-	if err := json.NewDecoder(httpRsp.Body).Decode(&stats); err != nil {
-		return fmt.Errorf("failed to json.Decode() result: %v", err)
-	}
-	got := stats.Logs[cfg.Prefix]
-	if got.LogID != want.LogID {
-		return fmt.Errorf("got stats.log-id %d, want %d", got.LogID, want.LogID)
+		// Accumulate per-server stats for this log into overall per-log stats.
+		got.HTTPAllReqs += gotSingle.HTTPAllReqs
+		for _, rc := range rcs {
+			got.HTTPAllRsps[rc] += gotSingle.HTTPAllRsps[rc]
```

Comment:
These are different statuses -- line 551 detects whether we fail to retrieve stats, whereas the range for all of the results that the server has served (e.g. it will have a 400 for the add-chain of a corrupt cert).